src/linagora.esn.calendar/app/calendar-configuration/calendar-configuration-tab-main/calendar-configuration-tab-main.controller.js

```diff
@@ -110,5 +111,19 @@ require('../../components/modals/calendar-delete-confirmation/calendar-delete-co
     function exportCalendar() {
       calendarService.exportCalendar(self.calendarToExport.calendarHomeId, self.calendarToExport.id);
     }
+    function _createSecretLinkkwithToken() {
+      const jwtPayload = {
+        calendarHomeId: self.calendarHomeId,
+        calendarId: self.calendar.id,
+        userId: session.user._id
+
+      };
+
+      calendarService.generateTokenForSecretLink(jwtPayload)
+        .then(resp => {
+          self.calendarSecretLink = `${window.location.origin}/calendar/api/calendars/secretLink?jwt=${resp.token}`;
+        });
```

Comment:
The service should return something like this:

```suggestion
        .then(token => {
          self.calendarSecretLink = `${window.location.origin}/calendar/api/calendars/secretLink?jwt=${token}`;
        });
```